### PR TITLE
Log seed for random

### PR DESCRIPTION
**Pull request by janisz**
See: https://github.com/janisz/bigcache/commit/e73c96d8b9c30e2f5247ce96521edbd62b5d096b/checks

### CODE REVIEW

**Comment by mxplusb** [comment]
Yeah, it would be a good start to working through the random queuing error. @janisz can you rebase?

**Comment by janisz** [comment]
I tried to reproduce it and it looks like it's independent from the seed. It looks like there is a problem with a concurrent aces to memory.

File: bigcache_test.go [hunk 1]
```diff
@@ -441,6 +441,11 @@ func TestCacheDel(t *testing.T) {
 // TestCacheDelRandomly does simultaneous deletes, puts and gets, to check for corruption errors.
 func TestCacheDelRandomly(t *testing.T) {
 	t.Parallel()
+
+	seed := time.Now().UTC().UnixNano()
+	rand.Seed(seed)
+	print("Seed: ", seed, "\n")
```
Comment (cristaloleg) [inline bigcache_test.go]:
```suggestion
	t.Printf("Seed: %v", seed)
```

Comment (janisz) [reply bigcache_test.go]: There is no such function as `t.Printf` we could use `t.Log` but it's not working where there is a panic

Comment (cristaloleg) [reply bigcache_test.go]: Sure, it's `Logf` https://godoc.org/testing#T.Logf

Comment (cristaloleg) [reply bigcache_test.go]: Well, but how `print` will in a case of a panic? maybe it's better to place print/log/whatever before `rand.Seed` ?

**Comment by janisz** [comment]
```diff
@@ -46,6 +48,10 @@ func readTimestampFromEntry(data []byte) uint64 {
 func readKeyFromEntry(data []byte) string {
        length := binary.LittleEndian.Uint16(data[timestampSizeInBytes+hashSizeInBytes:])
 
+       if headersSizeInBytes+int(length) > len(data) {
+               panic(fmt.Sprintf("\n%s\n", hex.Dump(data)))
+       }
+
        // copy on read
        dst := make([]byte, length)
        copy(dst, data[headersSizeInBytes:headersSizeInBytes+length])
```

```
go test -run TestCacheDelRandomly -v -count 100
=== RUN   TestCacheDelRandomly
=== PAUSE TestCacheDelRandomly
=== CONT  TestCacheDelRandomly
2019/12/02 19:20:42 Allocated new queue in 1.179µs; Capacity: 2308 
2019/12/02 19:20:42 Allocated new queue in 10.236µs; Capacity: 4616 
2019/12/02 19:20:42 Allocated new queue in 9.476µs; Capacity: 9232 
2019/12/02 19:20:42 Allocated new queue in 18.23µs; Capacity: 18464 
2019/12/02 19:20:42 Allocated new queue in 43.117µs; Capacity: 36928 
2019/12/02 19:20:42 Allocated new queue in 37.135µs; Capacity: 73856 
2019/12/02 19:20:42 Allocated new queue in 90.396µs; Capacity: 147712 
2019/12/02 19:20:42 Allocated new queue in 113.411µs; Capacity: 295424 
2019/12/02 19:20:42 Allocated new queue in 191.364µs; Capacity: 590848 
2019/12/02 19:20:42 Allocated new queue in 351.637µs; Capacity: 1048576 
panic: 
00000000  6e 6e 6e 6e 6e 6e 6e 6e  6e 6e 6e 6e 6e 6e 6e 6e  |nnnnnnnnnnnnnnnn|
00000010  6e 6e 6e 6e 65 6b 65 79  31 38 34 b8 b8 b8 b8 b8  |nnnnekey184.....|
00000020  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000030  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000040  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000050  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000060  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000070  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000080  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000090  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000000a0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000000b0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000000c0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000000d0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000000e0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000000f0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000100  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000110  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000120  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000130  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000140  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000150  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000160  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000170  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000180  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000190  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000001a0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000001b0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000001c0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000001d0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000001e0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000001f0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000200  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000210  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000220  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000230  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000240  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000250  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000260  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000270  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000280  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000290  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000002a0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000002b0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000002c0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000002d0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000002e0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000002f0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000300  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000310  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000320  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000330  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000340  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000350  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000360  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000370  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000380  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000390  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000003a0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000003b0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000003c0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000003d0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000003e0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
000003f0  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000400  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8 b8 b8 b8 b8 b8  |................|
00000410  b8 b8 b8 b8 b8 b8 b8 b8  b8 b8 b8                 |...........|



goroutine 23 [running]:
github.com/allegro/bigcache/v2.readKeyFromEntry(0xc000343c32, 0x41b, 0x643ce, 0xec95a236df9d09a2, 0xc000343c32)
	/home/janisz/go/src/github.com/allegro/bigcache/encoding.go:52 +0x231
github.com/allegro/bigcache/v2.(*cacheShard).get(0xc0000f2000, 0xc00044c580, 0x9, 0xec95a236df9d09a2, 0xc00044c580, 0x9, 0x9, 0xc00044c580, 0xc0000b20d0)
	/home/janisz/go/src/github.com/allegro/bigcache/shard.go:69 +0xcd
github.com/allegro/bigcache/v2.(*BigCache).Get(0xc000092ea0, 0xc00044c580, 0x9, 0x1, 0x1, 0xc00044c580, 0x9, 0x0)
	/home/janisz/go/src/github.com/allegro/bigcache/bigcache.go:118 +0x8b
github.com/allegro/bigcache/v2.TestCacheDelRandomly.func3(0xc3500, 0xc000092ea0, 0xc0000dc100, 0xc0000a0110)
	/home/janisz/go/src/github.com/allegro/bigcache/bigcache_test.go:491 +0x96
created by github.com/allegro/bigcache/v2.TestCacheDelRandomly
	/home/janisz/go/src/github.com/allegro/bigcache/bigcache_test.go:482 +0x294
exit status 2
FAIL	github.com/allegro/bigcache/v2	1.059s

```
It looks like two entries can override each other.

```
go version
go version go1.13.4 linux/amd64
```

**Pull request closed without merging**